File: backend/src/main/java/com/gzmtu/backend/entity/Book.java
/*
 * @Author: error: error: git config user.name & please set dead value or install git && error: git config user.email & please set dead value or install git & please set dead value or install git
 * @Date: 2025-06-21 16:18:07
 * @LastEditors: error: error: git config user.name & please set dead value or install git && error: git config user.email & please set dead value or install git & please set dead value or install git
 * @LastEditTime: 2025-06-21 16:40:15
 * @FilePath: \AAA\backend\src\main\java\com\gzmtu\backend\entity\Book.java
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 */
package com.gzmtu.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.sql.Timestamp;

@Data
@Entity
@Table(name = "book")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String title;
    private String author;
    private String publisher;
    private String isbn;
    private Double price;
    private Integer stock;
    private String description;
    private String cover;
    @Column(name = "category_id")
    private Integer categoryId;
    @Column(columnDefinition = "varchar(255) default 'on'")
    private String status;
    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;
} 

File: backend-go/model/book.go
package model

import "time"

// Book corresponds to the `book` table.
type Book struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Title       string    `gorm:"type:varchar(200);not null" json:"title"`
	Author      string    `gorm:"type:varchar(100);not null" json:"author"`
	Publisher   string    `gorm:"type:varchar(100)" json:"publisher,omitempty"`
	Isbn        string    `gorm:"type:varchar(20);unique" json:"isbn,omitempty"`
	Price       float64   `gorm:"type:decimal(10,2);not null" json:"price"`
	Stock       int       `gorm:"default:0" json:"stock"`
	Description string    `gorm:"type:text" json:"description,omitempty"`
	ImageURL    string    `gorm:"column:image_url;type:varchar(500)" json:"cover,omitempty"`
	Status      string    `gorm:"type:varchar(20);default:'on'" json:"status,omitempty"`
	CategoryID  uint      `gorm:"column:category_id" json:"categoryId"`
	CreatedAt   time.Time `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt   time.Time `gorm:"column:updated_at" json:"updatedAt"`
}

func (Book) TableName() string {
	return "book"
}
